library/alloc/src/collections/btree/node.rs

```diff
@@ -1796,5 +1694,42 @@ unsafe fn slice_remove<T>(slice: &mut [MaybeUninit<T>], idx: usize) -> T {
     }
 }
 
+/// Shifts the slice such that the element at `mid` becomes the first.
+/// Is similar to `rotate_left`, but assumes the leading elements are
+/// uninitialized and leaves the trailing elements uninitialized.
```

Comment:
I think the comparison to rotate_left is confusing. This does not perform any sort of rotation, it just shifts elements to the left. slice_shift_left might be a more appropriate name, and `mid` could be `distance` in that case.

Reply:
Ok, I meant it could be implemented using `rotate_left`.